### ISSUE TITLE
`mantil watch` never ends

### ISSUE THREAD

**Issue by ianic** [mantil-io/mantil#106]
In 0.2.5 when I start watch after the first change it is constantly detecting changes. We are creating build folder now, and there we create some Go files which are immediately triggering a new change. 
I [fix it now](https://github.com/mantil-io/mantil/commit/9591eb5c7db0b825c2ad3f16c12652b9c723c17b) by watching only api folder.

**Comment by IvanVlasic** [comment]
Watch command now watches everything inside the project directory.

I've tried fixing the issue through current watcher library with provided `Ignore` and `Remove` methods but that proved to be a challenge due to the way library works internally and combining that option with additional filters.

I've also tried to do it through the regex filters by adding a filter that matches all go files except `main.go` but that also proved to be a challenge due to Go not supporting this feature as there's no way to do it in O(n) complexity which is current library guarantee.

I also haven't managed to find decent replacement. The closest one is [fsnotify](https://github.com/fsnotify/fsnotify) but it doesn't support recursively adding folders which is necessary in our case.

Having tried all this, the solution I decided to implement is just to check if the file that triggered the change is `main.go` and ignoring the event in that case. So currently, during each deploy watch event will be triggered twice - first event will deploy the changes and then another event will be generated since deploy process created new `main.go` but that second event will be ignored by the Mantil.
